src/common.h

```diff
@@ -0,0 +1,257 @@
+    {0xc6e7a6ffb08e3363L, 0xe08fec7c86389beeL, 0xf2d38f10fbb8d1bbL, 0x0abe6a5e5abcaa32L},
+    {0x5616c57de0ec9eaeL, 0xc631ffb2585a72dbL, 0x5121af06a3b51e3cL, 0x73560252aa0655b2L},
+    {0x92cf4deb77bd779cL, 0x72cf6a8029b7d7bcL, 0x6e0bcd91ee762730L, 0x291cf6d68823e687L},
+    {0xce32ef844e11a51eL, 0xc0ba12bb3da64ca5L, 0x0454dc1edc61a1a3L, 0x019fe632fd328739L},
+    {0x531a11a0d2d75182L, 0x02c8118402867ddcL, 0x116168bffbedc11dL, 0x0a0a77a3b1980c0dL},
+    {0xe2d0a7869f0319edL, 0xb94f1101b1d7a628L, 0xece8ea224f31d25dL, 0x23397a9300f8f98bL},
+    {0xd7b688830a4f2089L, 0x6558e9e3f6ac7b41L, 0x99e276b571905a7dL, 0x52dd465e2f094256L},
+    {0x474650359d8e211bL, 0x84d37b826214abc6L, 0x8da40c1ef2bb4598L, 0x0c83ea7744bf1beeL},
+    {0x694341f608c9dd56L, 0xed3a181fabb30adcL, 0x1339a815da8b398fL, 0x2c6d4e4511657e1eL},
+    {0x63e7cb4906ffc93fL, 0xf070bb00e28a193dL, 0xad1715b02e5713b5L, 0x4b5371495990693fL}
+};
+
+////////////////////////////////////////////////////////////////////////////////////////////////////
+// Public Functions
+////////////////////////////////////////////////////////////////////////////////////////////////////
+
+/*
+ * Memory Allocation:
+ */
+C_KZG_RET c_kzg_malloc(void **out, size_t size);
+C_KZG_RET c_kzg_calloc(void **out, size_t count, size_t size);
+C_KZG_RET new_g1_array(g1_t **x, size_t n);
+C_KZG_RET new_g2_array(g2_t **x, size_t n);
+C_KZG_RET new_fr_array(fr_t **x, size_t n);
+
+/*
+ * General Helper Functions:
+ */
+bool is_power_of_two(uint64_t n);
+int log2_pow2(uint32_t n);
+uint32_t reverse_bits(uint32_t n);
+C_KZG_RET bit_reversal_permutation(void *values, size_t size, uint64_t n);
+
+/*
+ * Conversion and Validation:
+ */
+void bytes_from_g1(Bytes48 *out, const g1_t *in);
+void bytes_from_bls_field(Bytes32 *out, const fr_t *in);
+void bytes_from_uint64(uint8_t out[8], uint64_t n);
+C_KZG_RET bytes_to_bls_field(fr_t *out, const Bytes32 *b);
+C_KZG_RET bytes_to_kzg_commitment(g1_t *out, const Bytes48 *b);
+C_KZG_RET bytes_to_kzg_proof(g1_t *out, const Bytes48 *b);
+void fr_from_uint64(fr_t *out, uint64_t n);
+void hash_to_bls_field(fr_t *out, const Bytes32 *b);
+C_KZG_RET blob_to_polynomial(fr_t *p, const uint8_t *blob, size_t num_fields);
+
+/*
+ * Field Operations:
+ */
+bool fr_equal(const fr_t *a, const fr_t *b);
+bool fr_is_one(const fr_t *p);
+void fr_div(fr_t *out, const fr_t *a, const fr_t *b);
+void fr_pow(fr_t *out, const fr_t *a, uint64_t n);
+void compute_powers(fr_t *out, const fr_t *x, uint64_t n);
+
+/*
+ * Point Operations:
+ */
+void g1_sub(g1_t *out, const g1_t *a, const g1_t *b);
+void g1_mul(g1_t *out, const g1_t *a, const fr_t *b);
+bool pairings_verify(const g1_t *a1, const g2_t *a2, const g1_t *b1, const g2_t *b2);
+void g1_lincomb_naive(g1_t *out, const g1_t *p, const fr_t *coeffs, uint64_t len);
+C_KZG_RET g1_lincomb_fast(g1_t *out, const g1_t *p, const fr_t *coeffs, size_t len);
+C_KZG_RET g1_fft(g1_t *out, const g1_t *in, size_t n, const KZGSettings *s);
```

Comment:
Not sure if `g1_fft` (and `g1_ifft`) should be in common. We only use it in eip7594, right? 

On the other hand, it is a somewhat general operation that we may want to use elsewhere at some point.